pkg/client/client.go

```diff
@@ -109,20 +66,66 @@ func newClient(host string) *HTTPClient {
 		IdleConnTimeout:       90 * time.Second,
 		TLSHandshakeTimeout:   5 * time.Second,
 		ExpectContinueTimeout: 1 * time.Second,
+		DisableKeepAlives:     false,
 	}
-	transport.DisableKeepAlives = false
 	maxConnPerHost := viper.GetInt(optname.MaxConnPerHost)
 	if maxConnPerHost > 0 {
-		transport.MaxConnsPerHost = maxConnPerHost
+		baseTransport.MaxConnsPerHost = maxConnPerHost
 	}
 
-	client := &http.Client{
-		Transport:     &R8GetRetryingRoundTripper{Transport: transport},
-		CheckRedirect: checkRedirectFunc,
+	transport := &UserAgentTransport{Transport: &baseTransport}
+
+	retryClient := &retryablehttp.Client{
+		HTTPClient: &http.Client{
+			Transport:     transport,
+			CheckRedirect: checkRedirectFunc,
+		},
+		Logger:       nil,
+		RetryWaitMin: retryMinWait,
+		RetryWaitMax: retryMaxWait,
+		RetryMax:     viper.GetInt(optname.Retries),
+		CheckRetry:   retryablehttp.DefaultRetryPolicy,
+		Backoff:      linearJitterRetryAfterBackoff,
 	}
+
+	client := retryClient.StandardClient()
 	return &HTTPClient{Client: client, host: host}
 }
 
+// linearJitterRetryAfterBackoff wraps retryablehttp.LinearJitterBackoff but also will adhere to Retry-After responses
+func linearJitterRetryAfterBackoff(min, max time.Duration, attemptNum int, resp *http.Response) time.Duration {
+	var retryAfter time.Duration
+
+	if shouldApplyRetryAfter(resp) {
+		retryAfter = evaluateRetryAfter(resp)
+	}
+
+	if retryAfter > 0 {
+		// If the Retry-After header is set, treat this as attempt 0 to get just the jitter
+		attemptNum = 0
```

Comment:
```suggestion
		retryablehttp.LinearJitterBackoff(retryAfter, retryAfter + jitter, 0, resp)
```
where `jitter` is `max - min`.

The current solution adds `retryAfter` to `min` which means we wait a bit more than we ought to.